packages/sonataflow-deployment-webapp/src/routes/index.ts

```diff
@@ -0,0 +1,116 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+const IS_HASH_ROUTER = true;
+
+export enum QueryParams {}
+
+export enum PathParams {
+  EXTENSION = "extension",
+  FILE_PATH = "filePath",
+  MODEL_NAME = "modelName",
+}
+
+export class Route<
+  T extends {
+    pathParams?: any;
+    queryParams?: any;
+  }
+> {
+  constructor(private readonly pathDelegate: (pathParams: { [k in T["pathParams"]]: string }) => string) {}
+
+  public url(args: {
+    base?: string;
+    pathParams: { [k in T["pathParams"]]: string };
+    queryParams?: Partial<{ [k in T["queryParams"]]: string }>;
+  }) {
+    const SEP = args.base?.endsWith("/") ? "" : "/";
+    const HASH = IS_HASH_ROUTER ? "#" : "";
+    const path = this.pathDelegate(args.pathParams);
+    const queryParams = args.queryParams ?? {};
+
+    if (!args.base && Object.keys(queryParams).length <= 0) {
+      return `${HASH}${path}`;
+    }
+
+    if (!args.base) {
+      return `${HASH}${path}?${this.queryString(queryParams)}`;
+    }
+
+    if (Object.keys(queryParams).length <= 0) {
+      return `${args.base}${SEP}${HASH}${path}`;
+    }
+
+    return `${args.base}${SEP}${HASH}${path}?${this.queryString(queryParams)}`;
+  }
+
+  public queryString(queryParams: Partial<{ [k in T["queryParams"]]: string }>) {
+    return decodeURIComponent(new URLSearchParams(queryParams as Record<string, string>).toString());
+  }
+
+  public queryArgs(queryString: QueryParamsImpl<string>): QueryParamsImpl<T["queryParams"]> {
+    return queryString;
+  }
+
+  public path(pathParams: { [k in T["pathParams"]]: string }) {
+    return this.pathDelegate(pathParams);
+  }
+}
+
+export interface QueryParamsImpl<Q extends string> {
+  has(name: Q): boolean;
+  get(name: Q): string | undefined;
+  with(name: Q, value: string): QueryParamsImpl<Q>;
+  without(name: Q): QueryParamsImpl<Q>;
+  toString(): string;
+}
+
+export function newQueryParamsImpl<Q extends string>(queryString: string): QueryParamsImpl<Q> {
+  return {
+    has: (name) => new URLSearchParams(queryString).has(name),
+    get: (name) => {
+      const val = new URLSearchParams(queryString).get(name);
+      return !val ? undefined : decodeURIComponent(val);
+    },
+    with: (name, value) => {
+      const urlSearchParams = new URLSearchParams(queryString);
+      urlSearchParams.set(name, value);
+      return newQueryParamsImpl(decodeURIComponent(urlSearchParams.toString()));
+    },
+    without: (name) => {
+      const urlSearchParams = new URLSearchParams(queryString);
+      urlSearchParams.delete(name);
+      return newQueryParamsImpl(decodeURIComponent(urlSearchParams.toString()));
+    },
+    toString: () => {
+      return decodeURIComponent(new URLSearchParams(queryString).toString());
+    },
+  };
+}
+
+export const routes = {
+  home: new Route<{}>(() => "/"),
+  dataJson: new Route<{}>(() => "/data.json"),
```

Comment:
Can we have a more informative name for this file? As`data.json` is too generic, I would like to avoid any possible collision with others. On Dashbuilder viewer, for example, the name is `dashbuilder-viewer-deployment-webapp-data.json`